ompi/mca/mtl/psm2/mtl_psm2_endpoint.h

```diff
@@ -58,7 +58,13 @@ OBJ_CLASS_DECLARATION(mca_mtl_psm2_endpoint);
 static inline mca_mtl_psm2_endpoint_t *ompi_mtl_psm2_get_endpoint (struct mca_mtl_base_module_t* mtl, ompi_proc_t *ompi_proc)
 {
     if (OPAL_UNLIKELY(NULL == ompi_proc->proc_endpoints[OMPI_PROC_ENDPOINT_TAG_MTL])) {
-	ompi_mtl_psm2_add_procs (mtl, 1, &ompi_proc);
+        if (OPAL_UNLIKELY(OMPI_SUCCESS != MCA_PML_CALL(add_procs(&ompi_proc, 1)))) {
+            /* Fatal error. exit() out */
+            opal_output(0, "%s:%d: *** The Open MPI PSM2 MTL is aborting the MPI job (via exit(3)).\n",
+                           __FILE__, __LINE__);
+            fflush(stderr);
```

Comment:
FWIW, `opal_output()` uses an out-of-band mechanism to send the output back to `mpirun` (etc.).  Calling `fflush()` here may not have any effect.

Additionally, if you want to print a detailed error message, `opal_show_help()` is usually a better choice.

Reply:
Yeah, `fflush()` won't do anything as the output didn't go to `stderr`.

Reply:
So, I'm open to changing it, but I'd like to point out this is exactly the same code fragment that was approved for #7571 .